src/app/products/charges/create-charge/create-charge.component.html

```diff
@@ -0,0 +1,200 @@
+                Frequency Interval <strong>must be a positive integer</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field *ngIf="chargeForm.controls.chargeTimeType.value === 6 || chargeForm.controls.chargeTimeType.value === 7" fxFlex="48%">
+              <mat-label>Due Date</mat-label>
+              <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="dueDatePicker" required formControlName="feeOnMonthDay">
+              <mat-datepicker-toggle matSuffix [for]="dueDatePicker"></mat-datepicker-toggle>
+              <mat-datepicker #dueDatePicker></mat-datepicker>
+              <mat-error *ngIf="chargeForm.controls.feeOnMonthDay.hasError('required')">
+                Due Date is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="48%" *ngIf="chargeForm.controls.chargeTimeType.value === 7 || chargeForm.controls.chargeTimeType.value === 11">
+              <mat-label>Repeat Every ({{ repeatEveryLabel }})</mat-label>
+              <input type="number" step="1" matInput required formControlName="feeInterval">
+              <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('required')">
+                Repeat Every ({{ repeatEveryLabel }}) is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="(chargeForm.controls.feeInterval.hasError('min') || chargeForm.controls.feeInterval.hasError('max')) && !chargeForm.controls.feeInterval.hasError('pattern')">
+                Repeat Every (Months) <strong>must be between 1 and 12</strong>
+              </mat-error>
+              <mat-error *ngIf="chargeForm.controls.feeInterval.hasError('pattern')">
+                Repeat Every ({{ repeatEveryLabel }}) <strong>must be a positive integer</strong>
+              </mat-error>
+            </mat-form-field>
+  
+            <mat-form-field fxFlex="48%">
+              <mat-label>Amount</mat-label>
+              <input matInput required formControlName="amount">
+              <mat-error *ngIf="chargeForm.controls.amount.hasError('required') && !chargeForm.controls.amount.hasError('pattern')">
+                Amount is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="chargeForm.controls.amount.hasError('pattern') && currencyDecimalPoints !== 0 && currencyDecimalPoints !== undefined">
+                Amount must be <strong>a positive number with {{ currencyDecimalPoints }} or less decimal places</strong>
+              </mat-error>
+              <mat-error *ngIf="chargeForm.controls.amount.hasError('pattern') && currencyDecimalPoints === 0 && currencyDecimalPoints !== undefined">
+                Amount must be <strong>a positive number with no decimal places</strong>
+              </mat-error>
+              <mat-error *ngIf="chargeForm.controls.amount.hasError('pattern') && currencyDecimalPoints === undefined">
```

Comment:
3 Errors are necessary here. We do not want to say '0 or less decimal places' as that is simply bad UI. And we do not want to specify the decimal places before the user actually chooses the currency, but still has to warn the user that this value must be positive.